### PR TITLE
config de build para testes

### PR DESCRIPTION
**Pull request by igorgcosta**
### Descrição

Configurações de build da imagem docker da app no circleci, fazendo login no registry privado e fazendo push da imagem. Ao fim do push é enviada uma notificação com informações de status do build na página do PR.

### Features
- Build e push de imagem docker da app

### CODE REVIEW

**Comment by Dentxinho** [comment]
docker.sieve.com.br/b2wads/nodejs-boilerplate:8db52ebae456bfa7d650023513dcd72a8014b249

**Comment by igorgcosta** [comment]
:robot:

The app image was pushed to registry! :rocket:

Built with commit ```8db52``` :package:

Pull this image: ```docker.sieve.com.br/b2wads/nodejs-boilerplate:8db52```

**Comment by Dentxinho** [comment]
:robot:

The app image was pushed to registry! :rocket:

Built with commit [03b6102c3da7e9218dd6f822fd7949958c9daa6f](http://gooogle.com) :package:

Pull this image: **docker.sieve.com.br/b2wads/nodejs-boilerplate:03b6102c3da7e9218dd6f822fd7949958c9daa6f**

**Comment by Dentxinho** [comment]
:robot:

The app image was pushed to registry! :rocket:

Built with commit [66020](http://gooogle.com) :package:

Pull this image: **docker.sieve.com.br/b2wads/nodejs-boilerplate:66020**

**Comment by Dentxinho** [comment]
:robot:

The app image was pushed to registry! :rocket:

Built with commit [c9b0fa2](http://gooogle.com) :package:

Pull this image: **docker.sieve.com.br/b2wads/nodejs-boilerplate:c9b0fa2**

**Comment by Dentxinho** [comment]
:robot: The app image was pushed to registry!

   :package: Built with commit [9e5e47d](https://api.github.com/repos/b2wads/nodejs-boilerplate/issues/48/comments/commits/9e5e47d8f6b50ec4afd31aaf1cdc76b4212e6e94) 

   :rocket: Pull this image: **docker.sieve.com.br/b2wads/nodejs-boilerplate:9e5e47d**

**Comment by Dentxinho** [comment]
:robot: The app image was pushed to registry!

   :package: Built with commit [5a364ed](https://github.com/b2wads/nodejs-boilerplate/issues/48/comments/nodejs-boilerplate/pull/48/commits/5a364ed3a3e32bcb1eaa271febe6216085427eff) 

   :rocket: Pull this image: 
 - **docker.sieve.com.br/b2wads/nodejs-boilerplate:5a364ed** 
 - **docker.sieve.com.br/b2wads/nodejs-boilerplate:latest**

**Comment by Dentxinho** [comment]
:robot: The docker image was pushed to registry!

   :package: Built with commit [14e6ab3](https://github.com/b2wads/nodejs-boilerplate/pull/48/commits/14e6ab36cf31ff3ecd170b3a7aff5ef27280a862) 

   :rocket: Pull this image in: 
 - **docker.sieve.com.br/b2wads/nodejs-boilerplate:14e6ab3** 
 - **docker.sieve.com.br/b2wads/nodejs-boilerplate:latest**

**Comment by Dentxinho** [comment]
:robot: The docker image was pushed to registry!

   :package: Built with commit [774885b](https://github.com/b2wads/nodejs-boilerplate/pull/48/commits/774885bef185111464d93df40447eac2aa4ae9ad) 

   :rocket: Pull this image in: 
 - **docker.sieve.com.br/b2wads/nodejs-boilerplate:774885b** 
 - **docker.sieve.com.br/b2wads/nodejs-boilerplate:latest**

**Comment by Dentxinho** [comment]
:robot: The docker image was pushed to registry!

   :package: Built with commit [cabccd6](https://github.com/b2wads/nodejs-boilerplate/pull/48/commits/cabccd672fe97643c45dd999fca7bbc6cabf95ad) 

   :rocket: Pull this image in: 
 - **docker.sieve.com.br/b2wads/nodejs-boilerplate:cabccd6** 
 - **docker.sieve.com.br/b2wads/nodejs-boilerplate:latest**

**Comment by Dentxinho** [comment]
:robot: The docker image was pushed to registry!

   :package: Built with commit [55b0b53](https://github.com/b2wads/nodejs-boilerplate/pull/48/commits/55b0b53c952f8b4b93f080d4e46e5bcb9f643a85) 

   :rocket: Pull this image in: 
 - **docker.sieve.com.br/b2wads/nodejs-boilerplate:staging-feat-circleci-build-test** 
 - **docker.sieve.com.br/b2wads/nodejs-boilerplate:latest**

**Comment by Dentxinho** [comment]
> ocker.sieve.com.br/b2wads/nodejs-boilerplate:staging-feat-circleci-build-test

Mara

>  A staging docker image of this app was pushed to registry!
> 
>  Built with commit [f228b19](https://github.com/b2wads/nodejs-boilerplate/pull/48/commits/f228b19a8d810007c383e719810042183a2149a0)
> 
>  Pull this image in **docker.sieve.com.br/b2wads/nodejs-boilerplate:staging-feat-circleci-build-test**

Maravilindo!

**Comment by Dentxinho** [comment]
>  A staging docker image of this app was pushed to registry!
> 
>  Built with commit [f228b19](https://github.com/b2wads/nodejs-boilerplate/pull/48/commits/f228b19a8d810007c383e719810042183a2149a0)
> 
>  Pull this image in **docker.sieve.com.br/b2wads/nodejs-boilerplate:staging-feat-circleci-build-test**

Maravilindo!

**Comment by Dentxinho** [comment]
teste

**Comment by Dentxinho** [comment]
teste

**Comment by Dentxinho** [comment]
teste

**Comment by Dentxinho** [comment]
teste

File: .circleci/config.yml [hunk 1]
```diff
@@ -38,6 +41,75 @@ jobs:
...
+        type: string
+      github-repo-path:
+        type: string
+        default: $CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME
+      github-api-address:
+        type: string
+        default: https://api.github.com
+      github-address:
+        type: string
+        default: https://github.combuild
```
Comment (daltonmatos) [inline .circleci/config.yml]: Esse default é esse valor mesmo? Parece faltar uma `/` antes do `build`.

Comment (igorgcosta) [reply .circleci/config.yml]: é sem o build. Corrigido na próxima versão

**Comment by daltonmatos** [comment]
@igorgcosta acha que vale a pena documentar o uso dessa feature que esse PR implementa? Afinal, tem variaveis de ambiente e outras coisas que são necessárias para fazer funcionar em um registry privado, por exemplo.

Talvez valha começar essa doc na wiki mesmo, pra não "poluir" o README.

O que acha?

**Pull request merged**